io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/AbstractTableInstanceWriter.java

```diff
@@ -229,9 +236,48 @@ private void addProperty(List<String> headerRow, Map<String, Object> row, Object
 		if (!headerRow.contains(propertyTypeName)) {
 			headerRow.add(propertyTypeName);
 		}
+
+		if (property instanceof GeometryProperty<?>) {
+			GeometryProperty<?> geometryProperty = (GeometryProperty<?>) property;
+
+			Geometry geometry = geometryProperty.getGeometry();
+			property = simplifyGeometry(geometry);
+		}
+
 		row.put(propertyTypeName, property);
 	}
 
+	/**
+	 * Simplifies a MultiGeometry to its single component if it contains only
+	 * one element.
+	 * 
+	 * @param geometry The input geometry, expected to be a MultiPoint,
+	 *            MultiLineString, or MultiPolygon.
+	 * @return The simplified geometry, either a Point, LineString, Polygon, or
+	 *         the original MultiGeometry.
+	 */
+	public static Geometry simplifyGeometry(Geometry geometry) {
+		if (geometry instanceof MultiPoint) {
+			MultiPoint multiPoint = (MultiPoint) geometry;
+			if (multiPoint.getNumGeometries() == 1) {
+				return multiPoint.getGeometryN(0);
+			}
+		}
+		else if (geometry instanceof MultiLineString) {
+			MultiLineString multiLineString = (MultiLineString) geometry;
+			if (multiLineString.getNumGeometries() == 1) {
+				return multiLineString.getGeometryN(0);
+			}
+		}
+		else if (geometry instanceof MultiPolygon) {
+			MultiPolygon multiPolygon = (MultiPolygon) geometry;
+			if (multiPolygon.getNumGeometries() == 1) {
+				return multiPolygon.getGeometryN(0);
+			}
+		}
```

Comment:
I think the logic before with the `GeometryCollection` was fine, any specific reason to change it?

Reply:
Yes, you changed it. In the original version you did it the same way as in your changes to `InstanceToJson`, just checking for `GeometryCollection` and not these three subtypes. That's why I wondered if there is a specific reason to do it like that here.

Reply:
Please see the comments above related to your question.

Reply:
I can't really follow you. What proposal do you mean?

I see in the current commit you changed the logic back to checking `GeometryCollection` instead of the three subclasses that were checked in the code related to the original comment in this thread. The only thing I did was asking about if there was a specific reason for the change to check the three subclasses instead of only `GeometryCollection` as you did in your first implementation.

Reply:
No specific reason. In my opinion was a good approach but just using getGeometryN(0) it's enough for all the cases so I dropped the method.